src/tasks/WasmAppBuilder/WebcilConverter.cs

```diff
@@ -0,0 +1,44 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Collections.Immutable;
+using System.Reflection.PortableExecutable;
+
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.WebAssembly.Build.Tasks;
+
+/// <summary>
+/// Reads a .NET assembly in a normal PE COFF file and writes it out as a Webcil file
+/// </summary>
+public class WebcilConverter
+{
+    private readonly string _inputPath;
+    private readonly string _outputPath;
+
+    private readonly NET.WebAssembly.Webcil.WebcilConverter _converter;
+
+    private TaskLoggingHelper Log { get; }
+    private WebcilConverter(NET.WebAssembly.Webcil.WebcilConverter converter, string inputPath, string outputPath, TaskLoggingHelper logger)
+    {
+        _converter = converter;
+        _inputPath = inputPath;
+        _outputPath = outputPath;
+        Log = logger;
+    }
+
+    public static WebcilConverter FromPortableExecutable(string inputPath, string outputPath, TaskLoggingHelper logger)
+    {
+        var converter = NET.WebAssembly.Webcil.WebcilConverter.FromPortableExecutable(inputPath, outputPath);
+        return new WebcilConverter(converter, inputPath, outputPath, logger);
+    }
+
+    public void ConvertToWebcil()
+    {
+        Log.LogMessage($"Converting to Webcil: input {_inputPath} output: {_outputPath}");
```

Comment:
Do we want this to be visible to the user in the normal case? If not, we should use `Log.LogMessage(MessageImportance.Log, $"..`

Reply:
And we could add a user visible message somewhere in WasmAppBuilder saying `Converting assemblies to WebCil`.

Feedback can be addressed, if you choose to, in a follow up PR.